lib/Attachment.php

```diff
@@ -156,6 +156,53 @@ public function __toString() {
 	 * @param int|mixed $iid An attachment identifier.
 	 */
 	public function init( $iid = null ) {
+		$iid = $this->determine_id( $iid );
+
+		/**
+		 * determine_id returns null when the attachment is a file path,
+		 * thus there's nothing in the DB for us to do here 
+		 */
+		if ( null === $iid ) {
+			return null;
+		}
+
+		$attachment_info = $this->get_attachment_info( $iid );
+
+		$this->import( $attachment_info );
+
+		$basedir = self::wp_upload_dir();
+		$basedir = $basedir['basedir'];
+
+		if ( isset( $this->file ) ) {
+			$this->file_loc = $basedir . DIRECTORY_SEPARATOR . $this->file;
+		} elseif ( isset( $this->_wp_attached_file ) ) {
+			$this->file     = $this->_wp_attached_file;
+			$this->file_loc = $basedir . DIRECTORY_SEPARATOR . $this->file;
+		}
+
+		if ( isset( $attachment_info['id'] ) ) {
+			$this->ID = $attachment_info['id'];
+		} elseif ( is_numeric( $iid ) ) {
+			$this->ID = $iid;
+		}
+
+		if ( isset( $this->ID ) ) {
+			$this->import_custom( $this->ID );
+
+			$this->id = $this->ID;
+		}
+
+		return null;
+	}
+
+	/**
+	 * Tries to figure out the attachment id you want or otherwise handle when
+	 * a string or other data is sent (object, file path, etc.)
+	 * @internal
+	 * @param mixed a value to test against
+	 * @return int|null the numberic id we should be using for this post object 
+	 */
+	protected function determine_id( $iid ) {
```

Comment:
Should the `determine_id()` method also check for an instance of `WP_Post`? It could very well be a WordPress attachment we try to init with, right?

```php
/**
 * If passed a Timber\Attachment or WP_Post object, grab the ID and continue. Otherwise, try
 * to check for an ACF image array an take the ID from that array.
 */
if ( $iid instanceof Attachment
    || ( $iid instanceof \WP_Post && 'attachment' === $iid->post_type )
) {
    $iid = (int) $iid->ID;
} elseif ( is_array( $iid ) && isset( $iid['ID'] ) ) {
    // Assume ACF image array.
    $iid = $iid['ID'];
}
```

Reply:
In my opinion no. It is too forgiving for the user as it doesn't make sense to ouput a post as an image. Any thoughts?

Reply:
@pascalknecht I think the scenario @gchtr outlines is less about outputting a post as an image, but rather instances when the data passed to the function was for some reason sent inside a `WP_Post` class (for example, as the result of a `WP_Query`)